pyvrp/cpp/DurationSegment.h

```diff
@@ -139,18 +139,26 @@ DurationSegment DurationSegment::merge(Matrix<Duration> const &durationMatrix,
 
     // Time warp increases if we arrive at the other's first client after its
     // time window closes, whereas wait duration increases if we arrive there
-    // before opening.
-    Dur const diffTw = std::max<Dur>(twEarly_ + atOther - other.twLate_, 0);
+    // before opening. Ternaries below are used to avoid overflows.
+    Dur const diffTw = twEarly_ + atOther > other.twLate_
+                           ? twEarly_ + atOther - other.twLate_
+                           : 0;
     Dur const diffWait = other.twEarly_ - atOther > twLate_
                              ? other.twEarly_ - atOther - twLate_
-                             : 0;  // ternary rather than max avoids underflow
+                             : 0;
+
+    // Only add atOther to other.twLate_ if it does not result in overflow.
+    Dur const maxDur = std::numeric_limits<Dur>::max();
+    Dur const twLateAtOther = atOther >= other.twLate_ - maxDur
+                                  ? other.twLate_ - atOther
+                                  : other.twLate_;
```

Comment:
I had this before, but it didn't work. I assume that there are enough cases with infeasible solutions where `atOther` is negative.

```cpp
    Dur const twLateAtOther
        = atOther >= 0 ? other.twLate_ - atOther : other.twLate_;
```